app/models/leaderboard_row.rb

```diff
@@ -44,6 +49,15 @@ def weak_side_wins
     @corp_wins > @runner_wins ? @runner_wins : @corp_wins
   end
 
+  def add_stat(key, value)
+    @extra[key] = value
+  end
+
+  def method_missing(name)
+    return @extra[name] if @extra.has_key?(name)
+    super
+  end
+
```

Comment:
@richdouglasevans I agree that SOS is pretty central in Netrunner - and indeed many league systems - but I'd like to abstract out some of this complexity. For the tiebreakers, I'm currently defining separate Ranker classes that know only about how to rank two things based on their own specific metric. And calculating SOS is really a concern of the SOS ranker. It's pretty close though.

In fact, the new SOS that's coming round the corner will be calculated slightly differently, and structuring it like this allows the two systems to exist in parallel and isolated into their own classes instead of that complexity being in here.

I actually plan to refactor a lot of this stuff since I'm not altogether happy with all the param passing going on, and I'll think on it then. Hopefully it will look cleaner at least!